ydb/core/tablet_flat/flat_stat_table.h

```diff
@@ -140,40 +140,38 @@ class TKeyAccessSample {
         ui64 idx = TotalCount;
         ++TotalCount;
         if (idx >= SampleCount) {
-            idx = RandomNumber<ui64>(TotalCount) ;
+            idx = RandomNumber<ui64>(TotalCount);
         }
 
         if (idx >= SampleCount) {
             return;
         }
 
-        TSerializedCellVec saved(TSerializedCellVec::Serialize(key));
+        TString serializedKey = TSerializedCellVec::Serialize(key);
 
-        auto it = KeyRefCount.find(saved.GetBuffer());
-        if (it != KeyRefCount.end()) {
+        auto [it, inserted] = KeyRefCount.emplace(serializedKey, 1);
+        if (!inserted) {
             // Add a reference for existing key
-            saved = it->second.first;
-            ++it->second.second;
-        } else {
-            KeyRefCount[saved.GetBuffer()] = std::make_pair(saved, 1);
+            ++it->second;
         }
 
         if (Sample.size() < SampleCount) {
-            Sample.emplace_back(std::make_pair(saved.GetBuffer(), accessKind));
-        } else {
-            TString old = Sample[idx].first;
-            auto oit = KeyRefCount.find(old);
-            Y_VERIFY(oit != KeyRefCount.end());
-
-            // Delete the key if this was the last reference
-            if (oit->second.second == 1) {
-                KeyRefCount.erase(oit);
-            } else {
-                --oit->second.second;
-            }
+            Sample.emplace_back(std::make_pair(serializedKey, accessKind));
+            return;
+        }
+
+        TString old = Sample[idx].first;
+        auto oit = KeyRefCount.find(old);
+        Y_VERIFY(oit != KeyRefCount.end());
 
-            Sample[idx] = std::make_pair(saved.GetBuffer(), accessKind);
+        // Delete the key if this was the last reference
+        if (oit->second == 1) {
```

Comment:
Could be simplified to `if (0 == --oit->second) { KeyRefCount.erase(oit); }`. Comparions with `0` usually have more mechanical sympathy.

Reply:
Updated.